### ISSUE TITLE
[BUG] Default boundary condition is periodic even when specified otherwise

### ISSUE THREAD

**Issue by dhruvbalwada** [xgcm/xgcm#604]
The issue: Even when xgcm has been told that a coordinate is not periodic, it still chooses that as the default behavior - instead of throwing an error or warning. 

```
import xarray as xr 
import xgcm
import numpy as np

dx = 0.5
L = 20
x_left = np.arange(0,L-dx, dx)
x = np.arange(dx, L, dx)

slope = 2.
y = slope*x

da_y = xr.DataArray(y, dims=['xc'], coords={'xc':x})
ds_y = xr.Dataset(coords={'xc':x, 'xg':x_left})
ds_y['y'] = da_y

grid = xgcm.Grid(ds_y, 
                 coords={"X": {"center": "xc", "left": "xg"}},
                periodic=None)

dy_default = grid.diff(ds_y.y, 'X')
dy = grid.diff(ds_y.y, 'X', boundary='extend')
```

In this example we told grid that `X` is not a periodic direction. When we print grid, it says that `X` is not periodic. However, the dy_default (where we did not pass a boundary condition) still assumes `X` is periodic without alerting the user.

**Comment by dhruvbalwada** [comment]
Might be related to #509

**Comment by hdrake** [comment]
Confirming I'm running into this same issue.

**Comment by jbusecke** [comment]
Is this fixed by using the `boundary=` input syntax? In that case deprecation of `periodic` might be the solution here.

**Comment by hdrake** [comment]
> Is this fixed by using the `boundary=` input syntax? In that case deprecation of `periodic` might be the solution here.

Yep–see https://github.com/xgcm/xgcm/issues/606#issuecomment-1702742283.

But that's not in the most recent release `v0.8.1` yet, right?

**Comment by hdrake** [comment]
Revisiting this on current `master`.

**Status of the reported reproducer.** The specific cases that originally silently wrapped no longer do. Building a `Grid` with `periodic=None`, `periodic=False`, or an explicit `boundary=...` now yields a non-periodic (`fill`/`extend`) axis, so `grid.diff(..., 'X')` without a per-call `boundary` no longer wraps:

```python
for kw in [dict(periodic=False), dict(periodic=None), dict(boundary='fill')]:
    g = xgcm.Grid(ds_y, coords={"X": {"center": "xc", "left": "xg"}},
                  autoparse_metadata=False, **kw)
    print(kw, "->", g.axes['X'].boundary)   # all -> 'fill'
```

**What remains.** The only path that still defaults to periodic wrapping is the *documented default* `Grid(..., periodic=True)` (the constructor default) used when neither `periodic` nor `boundary` is supplied. That default already emits a `DeprecationWarning` pointing users to `boundary='periodic'`.

**Recommendation.** Flipping the global default away from periodic (or requiring an explicit boundary/padding) is a breaking API change that is already the subject of the in-progress migration — deprecating `periodic` (#626) and renaming `boundary` -> `padding` (#696). It is best landed there as a coordinated, deprecation-cycle change rather than as an isolated bugfix, so I'm not opening a separate PR for this. This also covers #509. Closing-as-tracked-by-#626/#696 would be reasonable once that migration completes.
